src/ring_theory/chain_of_divisors.lean

```diff
@@ -248,3 +313,113 @@ begin
     ←pow_dvd_iff_le_multiplicity],
   exact pow_image_of_prime_by_factor_order_iso_dvd hn hp d (pow_multiplicity_dvd _),
 end
+
+lemma multiplicity_prime_eq_multiplicity_image_by_factor_order_iso [decidable_eq (associates M)]
+  {m p : associates M} {n : associates N} (hn : n ≠ 0)
+  (hp : p ∈ normalized_factors m) (d : {l : associates M // l ≤ m} ≃o {l : associates N // l ≤ n}):
+  multiplicity p m = multiplicity ↑(d ⟨p, dvd_of_mem_normalized_factors hp⟩) n :=
+begin
+  refine le_antisymm (multiplicity_prime_le_multiplicity_image_by_factor_order_iso hp d) _,
+  suffices : multiplicity ↑(d ⟨p, dvd_of_mem_normalized_factors hp⟩) n ≤
+    multiplicity ↑(d.symm (d ⟨p, dvd_of_mem_normalized_factors hp⟩)) m,
+  { rw [d.symm_apply_apply ⟨p, dvd_of_mem_normalized_factors hp⟩, subtype.coe_mk] at this,
+    exact this },
+  letI := classical.dec_eq (associates N),
+  simpa only [subtype.coe_eta] using
+    (multiplicity_prime_le_multiplicity_image_by_factor_order_iso
+      (mem_normalized_factors_factor_order_iso_of_mem_normalized_factors hn hp d) d.symm),
+end
+
+end
+
+variables [unique (Mˣ)] [unique (Nˣ)]
+
+/-- The order isomorphism between the factors of `mk m` and the factors of `mk n` induced by a
+  bijection between the factors of `m` and the factors of `n` that preserves `∣`. -/
+@[simps]
+def mk_factor_order_iso_of_factor_dvd_equiv
+  {m : M} {n : N} (d : {l : M // l ∣ m} ≃ {l : N // l ∣ n}) (hd : ∀ l l',
+  ((d l) : N) ∣ (d l') ↔ (l : M) ∣ (l' : M)) :
+   {l : associates M // l ≤ associates.mk m} ≃o {l : associates N // l ≤ associates.mk n} :=
+{ to_fun := λ l, ⟨associates.mk (d ⟨associates_equiv_of_unique_units ↑l,
+    by obtain ⟨x, hx⟩ := l ; rw [subtype.coe_mk,  associates_equiv_of_unique_units_apply,
```

Comment:
I think the style is `blah; blah2`

Reply:
The style is `{blah, blah2}` unless you actualyl have multiple goals